tests/codegen/cucode/scripts/arrays.py

```diff
@@ -0,0 +1,32 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring
+#==============================================================================
+
+def double_loop_on_2d_array_C( z : 'int[:,:](order=C)' ):
+
+    from numpy import shape
+
+    m, n = shape( z )
+
+    for i in range( m ):
+        for j in range( n ):
+            z[i,j] = i-j
```

Comment:
I thought you didn't want to put arrays in v1 of the cuda stuff?

Reply:
it works just because it does inherit from `CCodePrinter`

Reply:
and as i didn't do any special handling of the arrays in cuda it works for `CPU`

Reply:
> it works 

Does this mean it prints or it runs in Cuda?

I think I don't know enough about Cuda. Can it also run CPU code with none of the restrictions that apply to GPU code?

Reply:
yeah you can run CPU code as you would do normally.
`Cuda` just adds the ability to use operation that will run on `GPU`.
you can still run you CPU code as always.